pkg/exchange/okex/parse.go

```diff
@@ -85,101 +92,44 @@ func (data *BookEvent) BookTicker() types.BookTicker {
+		Time:   data.MillisecondTimestamp.Time(),
 	}
 
-	for _, bid := range data.Bids {
-		book.Bids = append(book.Bids, types.PriceVolume{Price: bid.Price, Volume: bid.Volume})
+	for i := range data.Bids {
+		book.Bids = append(book.Bids, types.PriceVolume{Price: data.Bids[i][0], Volume: data.Bids[i][1]})
 	}
 
-	for _, ask := range data.Asks {
-		book.Asks = append(book.Asks, types.PriceVolume{Price: ask.Price, Volume: ask.Volume})
+	for j := range data.Asks {
+		book.Asks = append(book.Asks, types.PriceVolume{Price: data.Asks[j][0], Volume: data.Asks[j][1]})
 	}
 
 	return book
 }
 
-type BookEntry struct {
-	Price         fixedpoint.Value
-	Volume        fixedpoint.Value
-	NumLiquidated int
-	NumOrders     int
-}
-
-func parseBookEntry(v *fastjson.Value) (*BookEntry, error) {
-	arr, err := v.Array()
-	if err != nil {
-		return nil, err
-	}
-
-	if len(arr) < 4 {
-		return nil, fmt.Errorf("unexpected book entry size: %d", len(arr))
-	}
-
-	price := fixedpoint.Must(fixedpoint.NewFromString(string(arr[0].GetStringBytes())))
-	volume := fixedpoint.Must(fixedpoint.NewFromString(string(arr[1].GetStringBytes())))
-	numLiquidated, err := strconv.Atoi(string(arr[2].GetStringBytes()))
-	if err != nil {
-		return nil, err
-	}
-
-	numOrders, err := strconv.Atoi(string(arr[3].GetStringBytes()))
-	if err != nil {
-		return nil, err
-	}
-
-	return &BookEntry{
-		Price:         price,
-		Volume:        volume,
-		NumLiquidated: numLiquidated,
-		NumOrders:     numOrders,
-	}, nil
-}
-
-func parseBookData(v *fastjson.Value) (*BookEvent, error) {
-	instrumentId := string(v.GetStringBytes("arg", "instId"))
-	data := v.GetArray("data")
-	if len(data) == 0 {
-		return nil, errors.New("empty data payload")
-	}
-
-	// "snapshot" or "update"
-	action := string(v.GetStringBytes("action"))
-
-	millisecondTimestamp, err := strconv.ParseInt(string(data[0].GetStringBytes("ts")), 10, 64)
-	if err != nil {
+// Order book channel
+func parseBookData(v *WebSocketPushDataEvent) (*BookEvent, error) {
+	instrumentId := v.Arg.InstrumentID
+	data := v.Data
+	var bookEvent []BookEvent
+	if err := json.Unmarshal(data, &bookEvent); err != nil {
 		return nil, err
 	}
 
-	checksum := data[0].GetInt("checksum")
-
-	var asks []BookEntry
-	var bids []BookEntry
-
-	for _, v := range data[0].GetArray("asks") {
-		entry, err := parseBookEntry(v)
-		if err != nil {
-			return nil, err
-		}
-		asks = append(asks, *entry)
+	var action string
+	if v.Arg.Channel != string(WsChannelTypeBooks5) {
+		action = *v.Action
 	}
 
-	for _, v := range data[0].GetArray("bids") {
-		entry, err := parseBookEntry(v)
-		if err != nil {
-			return nil, err
-		}
-		bids = append(bids, *entry)
-	}
+	checksum := bookEvent[0].Checksum
 
 	return &BookEvent{
 		InstrumentID:         instrumentId,
 		Symbol:               toGlobalSymbol(instrumentId),
 		Action:               action,
-		Bids:                 bids,
-		Asks:                 asks,
+		Bids:                 bookEvent[0].Bids,
```

Comment:
Why you can only assign index 0 data?

Reply:
because its return is like below 
```
{
  "arg": {
    "channel": "books",
    "instId": "BTC-USDT"
  },
  "action": "update",
  "data": [
    {
      "asks": [],
      "bids": [
        [
          "26253.4",
          "0.40181684",
          "0",
          "2"
        ],
        [
          "26253.2",
          "0.34070782",
          "0",
          "2"
        ]
      ],
      "ts": "1695793361109",
      "checksum": 419662638,
      "seqId": 14559343387,
      "prevSeqId": 14559343383
    }
  ]
}
```
data is array format, but always return 1 data (data always has length 1 )